### PR TITLE
Update asgard:install to add a prompt for APP_URL

### PR DESCRIPTION
**Pull request by johnsvenn**
I've updated the install asgard:install script to prompt for the APP_URL variable in the .env file - a lot of people run multiple sites or versions on development servers and don't just use http://locahost - so it's just a case of making life a little easier.

In addition, I've done some refactoring to:

1. Make it easier to add prompts for additional .env variables in the future and update the .env file using the EnvFileWriter
2. Add additional information to the prompt displayed when Asgard is already installed.

I would be happy to apply these changes to the 3.0 branch if required too.

### CODE REVIEW

File: Modules/Core/Console/InstallCommand.php [hunk 1]
```diff
@@ -56,16 +56,18 @@ public function fire()
...
-            \Modules\Core\Console\Installers\Scripts\ProtectInstaller::class,
-            \Modules\Core\Console\Installers\Scripts\ConfigureDatabase::class,
-            \Modules\Core\Console\Installers\Scripts\SetAppKey::class,
-            \Modules\Core\Console\Installers\Scripts\ConfigureUserProvider::class,
+            \Modules\Core\Console\Installers\Scripts\ProtectInstaller::class, // check if the .env file already exists
+            \Modules\Core\Console\Installers\Scripts\CreateEnvFile::class, // create a .env file
+            \Modules\Core\Console\Installers\Scripts\ConfigureDatabase::class, // prompt for database connection details
+            \Modules\Core\Console\Installers\Scripts\SetAppKey::class, // generate a unqiue app key
+            \Modules\Core\Console\Installers\Scripts\ConfigureAppUrl::class, // prompt for the app url
+            \Modules\Core\Console\Installers\Scripts\ConfigureUserProvider::class, // configure the admin user
```
Comment (nWidart) [inline Modules/Core/Console/InstallCommand.php]: Please remove comments.

File: Modules/Core/Console/InstallCommand.php [hunk 2]
```diff
@@ -56,16 +56,18 @@ public function fire()
...
             \Modules\Core\Console\Installers\Scripts\ModuleMigrator::class,
             \Modules\Core\Console\Installers\Scripts\ModuleSeeders::class,
             \Modules\Core\Console\Installers\Scripts\ModuleAssets::class,
             \Modules\Core\Console\Installers\Scripts\ThemeAssets::class,
             \Modules\Core\Console\Installers\Scripts\UnignoreComposerLock::class,
-            \Modules\Core\Console\Installers\Scripts\SetInstalledFlag::class,
+            \Modules\Core\Console\Installers\Scripts\SetInstalledFlag::class, // set the INSTALLED=true flag
```
Comment (nWidart) [inline Modules/Core/Console/InstallCommand.php]: Comments

File: Modules/Core/Console/Installers/Scripts/ConfigureDatabase.php [hunk 3]
```diff
@@ -45,16 +45,19 @@ public function fire(Command $command)
...
-            $port = $this->askDatabasePort($driver);
-            $name = $this->askDatabaseName();
-            $user = $this->askDatabaseUsername();
-            $password = $this->askDatabasePassword();
+            
+            $vars['db_driver'] = $this->askDatabaseDriver();
+            $vars['db_host'] = $this->askDatabaseHost();
+            $vars['db_port'] = $this->askDatabasePort($vars['db_driver']);
+            $vars['db_database'] = $this->askDatabaseName();
+            $vars['db_username'] = $this->askDatabaseUsername();
```
Comment (nWidart) [inline Modules/Core/Console/Installers/Scripts/ConfigureDatabase.php]: Not related to your PR

File: Modules/Core/Console/Installers/Scripts/ConfigureDatabase.php [hunk 4]
```diff
@@ -141,20 +144,18 @@ protected function askDatabasePassword()
...
     {
+        $driver = $vars['db_driver'];
+        
         $this->config['database.default'] = $driver;
-        $this->config['database.connections.' . $driver . '.host'] = $host;
-        $this->config['database.connections.' . $driver . '.port'] = $port;
-        $this->config['database.connections.' . $driver . '.database'] = $name;
-        $this->config['database.connections.' . $driver . '.username'] = $user;
-        $this->config['database.connections.' . $driver . '.password'] = $password;
+        $this->config['database.connections.' . $driver . '.host'] = $vars['db_host'];
```
Comment (nWidart) [inline Modules/Core/Console/Installers/Scripts/ConfigureDatabase.php]: Not related to your PR

File: Modules/Core/Console/Installers/Scripts/ProtectInstaller.php [hunk 5]
```diff
@@ -31,7 +31,7 @@ public function __construct(Filesystem $finder)
     public function fire(Command $command)
     {
         if ($this->finder->isFile('.env') && ! $command->option('force')) {
-            throw new Exception('Asgard has already been installed. You can already log into your administration.');
+            throw new Exception('Asgard has already been installed. You can already log into your administration.' . PHP_EOL . 'To create force install and create a new .env file run: php artisan asgard:install -f');
```
Comment (nWidart) [inline Modules/Core/Console/Installers/Scripts/ProtectInstaller.php]: Not related to your PR

**Comment by johnsvenn** [comment]
Thanks for the feedback - sorry I'm not used to contributing to public projects.

If I redo this for the 3.0 branch and then split up the request as below would the following be acceptable in principle?

PR 1 - Refactor EnvFileWriter so that it can be used by other scripts in addition to ConfigureDatabase (and refactor existing installer scripts to use this where applicable)

PR 2 - Add prompt for APP_URL (using the refactored EnvFileWriter)

PR 3 - Additional information when Asgard is already installed

Cheers

**Comment by nWidart** [comment]
That sounds good, thanks! 👍

**Comment by johnsvenn** [comment]
You're welcome. Cheers

**Pull request closed without merging**